src/utils/__tests__/internalLinks-test.js

```diff
@@ -116,61 +131,85 @@ describe('isNarrowLink', () => {
   }
 });
 
-describe('getLinkType', () => {
-  test('links to a different domain are of "non-narrow" type', () => {
-    expect(getLinkType(new URL('https://google.com/some-path'), realm)).toBe('non-narrow');
-  });
+// TODO: Combine / dedupe with "getNarrowFromLink (more)" tests, below
+describe('getNarrowFromLink', () => {
+  const mkCheck = (narrowExpectation: (Narrow => boolean) | null) => hash => {
+    const streams = [
+      eg.makeStream({ name: 'jest' }),
+      eg.makeStream({ name: 'stream' }),
+      eg.makeStream({ name: 'topic' }),
+      eg.makeStream({ name: 'mobile' }),
+    ];
+    const baseState = eg.reduxStatePlus({ streams });
+    const narrow = getNarrowFromLink(
+      new URL(hash, realm),
+      realm,
+      getStreamsById(baseState),
+      getStreamsByName(baseState),
+      getOwnUserId(baseState),
+    );
 
-  test('only in-app link containing "stream" is a stream link', () => {
-    expect(getLinkType(new URL('/#narrow/pm-with/1,2-group', realm), realm)).toBe('pm');
```

Comment:
I think these 7 commits in the middle:
8057ee788 internalLinks tests: Rename a test to be about all PMs, not just group ones
60566f355 internalLinks tests: Remove duplicated `expect`s on PM links
d021d5c58 internalLinks tests: Remove duplicated `expect` on a stream link
8ae9c6eff internalLinks tests: Move a stream-link `expect` into stream-links section
ca403f21c internalLinks tests: Rename some tests with puzzling names
37886141c internalLinks tests: Remove duplicated `expect`s on topic links
4927d83aa internalLinks tests: Move `expect`s on 'home' into the test meant for those

may be clearest if squashed together, with a description like "Organize getLinkType cases by expected result" and mentioning that some cases were duplicated. The diffstat is
```
 src/utils/__tests__/internalLinks-test.js | 30 +++++++++++++++---------------
 1 file changed, 15 insertions(+), 15 deletions(-)
```

As you observe, the existing organization of these tests is kind of puzzling with some odd names, and some of the intermediate stages are puzzling as a result.